src/main/java/com/scaleunlimited/flinkcrawler/fetcher/commoncrawl/CommonCrawlFetcherBuilder.java

```diff
@@ -90,38 +92,28 @@ private AmazonS3 makeClient() {
                 .build();
 	}
 	
-	private File makeCacheFile() {
-		// TODO support using S3 as cache location.
-		if (_cacheDir == null) {
-			throw new IllegalStateException("Can't set up cache file if cache dir hasn't been set");
-		}
+	private File makeCacheFile(File cacheDir) {
 		if (_crawlId == null) {
 			throw new IllegalStateException("Can't set up cache file if crawl id hasn't been set");
 		}
-		File cacheDir = new File(_cacheDir);
 		return new File(cacheDir, String.format("%s-%s", _crawlId, SERIALIZED_SECONDARY_INDEX_FILENAME));
 	}
 	
 	@Override
 	public BaseHttpFetcher build() throws Exception {
 		AmazonS3 client = makeClient();
-
-		// If the caller hasn't set up a cacheDir then we create a temp location for it
-		if (_cacheDir == null) {
-			File tempDir = File.createTempFile("cache-dir", "");
-			tempDir.delete();
-			tempDir.mkdir();
-			_cacheDir = tempDir.toString();
-		}
 		
-		File cachedFile = makeCacheFile();
-		// If cachedFile doesn't already exist, then let prepCache do so.
-		if (!cachedFile.exists()) {
-			prepCache(cachedFile);
+		// If the caller hasn't set up a cachedFile then we create one in a temp location
+		if (_cachedFile == null) {
+            File tempDir = File.createTempFile("cache-dir", "");
```

Comment:
This creates a temp file, not a temp directory, yes? I expected you would be calling `File.createTempDirectory()`

Reply:
This same directory creation code was in the original source (previous hunk), but I guess I could rewrite everything to use Path instead of File so that I can use Files.createTempDirectory() instead.